server/autolinkplugin/plugin.go

```diff
@@ -125,6 +120,12 @@ func (p *Plugin) ProcessPost(c *plugin.Context, post *model.Post) (*model.Post,
 	conf := p.getConfig()
 	postText := post.Message
 	offset := 0
+
+	channelName, teamName, rsErr := p.resolveScope(post.ChannelId)
```

Comment:
What we could do here is only call `resolveScope` if there is at least one non empty scope in the `cond.Links`. Otherwise we do not need `channelName` and `teamName`.